STACpopulator/extensions/datacube.py

```diff
@@ -192,44 +193,47 @@ def variables(self) -> dict[str, Variable]:
             # Some variables like "time_bnds" in some model files do not have any attributes.
             attrs = meta.get("attributes", {})
 
-            self._infer_variable_units_description(name, attrs)
+            if name in bounds:
+                # Bounds are auxiliary variables
+                dtype = VariableType.AUXILIARY.value
+
+                # We can safely assume that the bounds variable has the same units as the variable it bounds.
+                if "units" not in attrs:
+                    if (u := self.attrs["variables"][bounds[name]].get("attributes", {}).get("units")) is not None:
+                        attrs["units"] = u
+
+            elif self.is_coordinate(attrs):
+                # Using the CF-xarray heuristics to determine if variable is a coordinate.
```

Comment:
Is this if condition ever entered, given the presence of the `continue` statement on line 191?

Reply:
Okay, I guess the two are complementary and not identical.

Reply:
Correct.